WordPress/Classes/Services/AccountService.h

```diff
@@ -136,6 +136,19 @@ extern NSNotificationName const WPAccountEmailAndDefaultBlogUpdatedNotification;
                             success:(nullable void (^)(void))success
                             failure:(nullable void (^)(NSError *error))failure;
 
+
+/**
+ Syncs the details for the account associated with the provided auth token, then
+ creates or updates a WPAccount with the synced information.
+
+ @param authToken The auth token associated with the account being created/updated.
+ @param success A success block.
+ @param failure A failure block.
+ */
+- (void)syncAccountDetailsAndCreateAccount:(NSString *)authToken
```

Comment:
Nitpicking names, but I'd propose just `createOrUpdateAccountWithAuthToken:success:failure:`. I think this method says "I give you an oAuth2 token, go create an account with that" and it's up to the service to figure out what details it needs to do that.